### PR TITLE
Interruptible node affinity and anti-affinity

### PR DESCRIPTION
**Pull request by jeevb**
# TL;DR
Use node affinity to specify node targeting for interruptible and non-interruptible tasks.

## Type
 - [ ] Bug Fix
 - [x] Feature
 - [ ] Plugin

## Are all requirements met?

 - [x] Code completed
 - [x] Smoke tested
 - [x] Unit tests added
 - [ ] Code documentation added
 - [ ] Any pending items have an associated Issue

## Complete description
 _How did you fix the bug, make the feature etc. Link to any design docs etc_

## Tracking Issue
_Remove the '*fixes*' keyword if there will be multiple PRs to fix the linked issue_

fixes https://github.com/flyteorg/flyte/issues/<number>

## Follow-up issue
_NA_
OR
_https://github.com/flyteorg/flyte/issues/<number>_

### CODE REVIEW

**Comment by jeevb** [comment]
Just wanted to get some feedback on this draft. Eventually we can drop `InterruptibleNodeSelector` entirely.

**Comment by jeevb** [comment]
This is smoke tested for now on my sandbox, but waiting on tests/comments until after there are some initial comments.

File: go/tasks/pluginmachinery/flytek8s/config/config.go [hunk 1]
```diff
@@ -90,6 +90,7 @@ type K8sPluginConfig struct {
 	// Node Selector Labels for interruptible pods: Similar to InterruptibleTolerations, these node selector labels are added for pods that can tolerate
 	// eviction.
 	InterruptibleNodeSelector map[string]string `json:"interruptible-node-selector" pflag:"-,Defines a set of node selector labels to add to the interruptible pods."`
```
Comment (jeevb) [inline go/tasks/pluginmachinery/flytek8s/config/config.go:92]: We can drop this in favor of `InterruptibleNodeLabel`

**Review by EngHabu — APPROVED** [review]
Thank you!

**Review by EngHabu — APPROVED** [review]
